src/qt/automintdialog.h

```diff
@@ -51,4 +51,44 @@ private Q_SLOTS:
     void ensureLelantusModel();
 };
 
+enum class AutoMintSparkMode : uint8_t {
+    MintAll, // come from overview page
+    AutoMintAll // come from notification
+};
+
+class AutoMintSparkDialog : public QDialog
```

Comment:
The same here, Just add there spark related thins and check when use Lelantus related parts when spark, no need to have separate Dialogs.